scipy/optimize/_trustregion_exact.py

```diff
@@ -0,0 +1,424 @@
+                    num = trust_radius**2-pnorm**2
+                    den = pdotz+np.sign(pdotz)*np.sqrt(pdotz**2+num)
+                    steplen = num/den
+
+                    # Compute the quadratic term  (p.T*H*p)
+                    quadratic_term = np.dot(p, np.dot(H, p))
+
+                    # Check stop criteria
+                    relative_error = steplen**2*smin**2/(quadratic_term +
+                                                         damp*trust_radius**2)
+                    if relative_error <= self.k_hard:
+                        p = p+steplen*zmin
+                        break
+
+                    # Update uncertanty bounds
+                    ub = damp
+                    lb = max(lb, damp-smin**2)
+
+                    # Compute Cholesky factorization
+                    H = self.hess+damp_next*np.eye(n)
+                    c, info = self.cholesky(H, lower=False, overwrite_a=False,
+                                            clean=True)
+
+                    # Check if the factorization have succeded
+                    #
+                    if info == 0:  # Successfull factorization
+                        # Update damping factor
+                        damp = damp_next
+                        already_factorized = True
+                    else:  # Unsuccessfull factorization
+                        # Update uncertanty interval
+                        lb = max(lb, damp_next)
+
+                        # Update damping factor
+                        damp = max(np.sqrt(lb*ub), lb+self.UPDATE_COEF*(ub-lb))
+
+                else:  # Outside boundary
+                    # Check stop criteria
+                    relative_error = abs(pnorm - trust_radius)/trust_radius
+                    if relative_error <= self.k_easy:
+                        break
+
+                    # Update uncertanty bounds
+                    lb = damp
+
+                    # Update damp
+                    damp = damp_next
+
+            elif info == 0 and self.jac_mag <= self.CLOSE_TO_ZERO:
+                # jac_mag very close to zero
+
+                smin, zmin = estimate_smallest_singular_value(c)
+                steplen = trust_radius
+
+                # Check stop criteria
+                if steplen**2*smin**2 <= self.k_hard*(damp*trust_radius**2):
+                    p = steplen*zmin
+                    break
+
+                # Check for interior convergence
+                if damp == 0:
+                    p = np.zeros(n)
+                    hits_boundary = False
+                    break 
+
+                # Update uncertanty bounds
+                ub = damp
+                lb = max(lb, damp-smin**2)
+
+                # Update damping factor
+                damp = max(np.sqrt(lb*ub), lb+self.UPDATE_COEF*(ub-lb))
+
+            else:  # Unsuccessfull factorization
+
+                # Compute auxiliar terms
+                delta, v = singular_leading_submatrix(H, c, info)
```

Comment:
The result I expect is being produced. I checked it. The need for this incomplete factorisation is actually the reason I used the Lapack implementation of the Cholesky decomposition 'potrf' instead of the ``scipy.linalg.cholesky``